### PR TITLE
Add uncertainty-based method RTS.

### PR DESCRIPTION
**Pull request by MouxiaoHuang**
Add implementation of "Improving Training and Inference of Face Recognition Models via Random Temperature Scaling": https://arxiv.org/abs/2212.01015, an uncertainty-based method for OOD detection.

### CODE REVIEW

**Review by JediWarriorZou — COMMENTED** [review]
It looks good and thanks for your contributions.

File: scripts/basics/cifar100/train_cifar100.sh [hunk 1]
```diff
@@ -15,4 +15,4 @@ python main.py \
 --config configs/datasets/cifar100/cifar100.yml \
 configs/preprocessors/base_preprocessor.yml \
 configs/networks/resnet18_32x32.yml \
-configs/pipelines/train/baseline.yml
+configs/pipelines/train/baseline.yml
```
Comment (Jingkang50) [inline scripts/basics/cifar100/train_cifar100.sh:18]: no newline at end of file

File: scripts/ood/msp/cifar100_test_ood_msp.sh [hunk 2]
```diff
@@ -19,4 +19,4 @@ configs/preprocessors/base_preprocessor.yml \
 configs/postprocessors/msp.yml \
 --num_workers 8 \
 --network.checkpoint 'results/checkpoints/cifar100_res18_acc77.10.ckpt' \
---mark 0
+--mark 0
```
Comment (Jingkang50) [inline scripts/ood/msp/cifar100_test_ood_msp.sh:22]: no newline at end of file

File: openood/evaluators/base_evaluator.py [hunk 3]
```diff
@@ -40,6 +40,8 @@ def eval_acc(self,
 
                 # forward
                 output = net(data)
+                if isinstance(output, tuple):
```
Comment (Jingkang50) [inline openood/evaluators/base_evaluator.py:43]: when the output would be a tuple? could we prevent it?

**Review by Jingkang50 — COMMENTED** [review]
Thank you @MouxiaoHuang so much for contributing!
Could you also add your method to necessary places in README?
Also, it would be great if you can write a few sentence to describe your method in this PR.

Thanks

File: openood/evaluators/base_evaluator.py (hunk 3 above)
Comment (MouxiaoHuang) [reply openood/evaluators/base_evaluator.py]: I have added my method to README.
RTS (random temperature scaling) is a method originally designed to estimate face uncertainty. It models data uncertainty through distributional estimation, utilizing learnable mean (feature) and variance (uncertainty). Interestingly, we discovered that the temperature scalar precisely corresponds to the scale of uncertainty noise implicitly added in the softmax function. RTS offers several advantages: 1) It adjusts the learning strength of both clean and noisy samples to enhance stability and accuracy during the training phase. 2) It provides a confidence score to detect input uncertainty. 3) In addition to the uncertainty score, the MSP score can be used to identify OOD data.

**Review by Jingkang50 — APPROVED** [review]

**Pull request merged**